api/models/datastore.go

```diff
@@ -5,21 +5,66 @@ import (
 	"errors"
 )
 
+//TODO path name validation: implementations currently assume/require/rely on:
+// - leading slash
+// - overall legal/scrubbed path (no relative paths, consecutive slashes, etc)
+// - trailing slash allowed and distinct from none
```

Comment:
No problem waiting for another issue, though both inserting and querying these sorts of routes could have very strange behavior in the mean time.  Especially characters significant to regexp. (those regexps may need some additional escaping depending on what is allowed)